packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,234 @@
+
+```ts
+class TableColumnFileSizeCellView extends TableColumnTextCellViewBase<
+    TableColumnFileSizeCellRecord,
+    TableColumnFileSizeColumnConfig
+> {
+    ...
+
+    private updateText(): void {
+        if (this.columnConfig) {
+            this.text = formatFileSize(
+                this.columnConfig.formatter, // instance of Intl.NumberFormat
+                this.unitStrings, // either ['byte', 'bytes', 'KB', ...] or ['byte', 'bytes', 'KiB', ...]
+                this.columnConfig.unitType,
+                this.cellRecord?.value
+            );
+        } else {
+            this.text = '';
+        }
+    }
+}
+
+// Shared function also used by group header view
+function formatFileSize(
+    formatter: Intl.NumberFormat,
+    unitStrings: string[],
+    unitType: FileSizeUnitType,
+    byteCount: number
+): string {
+    let currentSize = byteCount;
+    let unitIndex = 0;
+    if (currentSize !== 1) {
+        const divisor = unitType === FileSizeUnitType.binary ? 1024 : 1000;
+        while (currentSize >= divisor && unitIndex < unitStrings.length) {
+            currentSize /= divisor;
+            unitIndex += 1;
+        }
+    }
+
+    return `${formatter.format(currentSize)} ${unitStrings[unitIndex]}`;
+}
+```
+
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-file-size` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization of the number
+
+For this we will use `Intl.NumberFormat`, passing in the value of the `lang` design token as the locale.
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be a great API for getting a file size unit string for a given locale. The only implemenation I've found is `Intl.NumberFormat`. This function can also do the numeric conversion, but we would not use it for that, because it only supports converting by using a factor of 1000, not 1024. We could instead do the number conversion ourselves and just ask `Intl.NumberFormat` to translate the unit string we need. However, we would have to work around several limitations:
+
+-   Does not have translations for KiB/MiB/etc.
+-   Does not translate plural "bytes".
+-   Throws error for locales with lang subtag `zh_CN` and `zh_TW` -- expects just `zh`.
+-   Requires creating separate `Intl.NumberFormat` instances for each unit we wish to translate.
+-   Missing translations for "byte" in French (gives "o" instead of "octet"), Japanese ("バイト"), Chinese ("字节" in simplified or "位元組" in traditional), Korean ("바이트"), Russian ("Б" instead of "байт"), and probably others.
```

Comment:
I'm not convinced that there are actually missing translations. I think you just need to use `unitDisplay: 'long'` in the case that you set `unit: 'byte'`. Then, I believe you get all the expected strings, including the English distinction of `byte` vs `bytes` (which is called out as a problem a few bullet points above).

Reply:
This is great. You're right. Using `unitDisplay: 'long'` does the trick.